pulp_ansible/app/galaxy/v3/views.py

```diff
@@ -257,108 +277,64 @@ def get_queryset(self):
-        collections_qs = queryset.annotate(available_versions=ArrayAgg("versions__version"))
-        versions_context = {}
-        for collection_id, available_versions in collections_qs.values_list(
-            "pk", "available_versions"
-        ):
-            versions_context[collection_id] = available_versions
-
-        self.available_versions_context = versions_context  # needed by get__serializer_context
-        self._deprecation
-
-        collections = Collection.objects.filter(
-            pk__in=versions_context.keys(),
-            versions__version_memberships__repository=repo_version.repository,
-        ).annotate(
-            repo_version_added_at=Window(
-                expression=FirstValue(
-                    "versions__version_memberships__version_added__pulp_last_updated"
-                ),
-                partition_by=[F("versions__collection_id")],
-                order_by=F("versions__version_memberships__version_added__pulp_last_updated").desc(
-                    nulls_last=True
-                ),
-            ),
-            repo_version_removed_at=Window(
-                expression=FirstValue(
-                    "versions__version_memberships__version_removed__pulp_last_updated"
-                ),
-                partition_by=[F("versions__collection_id")],
-                order_by=F(
-                    "versions__version_memberships__version_removed__pulp_last_updated"
-                ).desc(nulls_last=True),
-            ),
+        deprecated_qs = filter_content_for_repo_version(
+            AnsibleCollectionDeprecated.objects, repo_version
+        ).filter(namespace=OuterRef("namespace"), name=OuterRef("name"))
+
+        latest_cv_version_qs = (
+            filter_content_for_repo_version(CollectionVersion.objects, repo_version)
+            .filter(collection=OuterRef("pk"))
+            .order_by(
+                "-version_major",
+                "-version_minor",
+                "-version_patch",
+                "-version_prerelease",
+                "-pulp_created",
+            )
+            .only("version")
         )
 
-        return collections.distinct("versions__collection_id").only(
-            "pulp_created", "name", "namespace"
+        latest_version_modified_qs = (
+            RepositoryContent.objects.filter(
+                repository_id=repo_version.repository_id,
+                version_added__number__lte=repo_version.number,
+            )
+            .select_related("content__ansible_collectionversion")
+            .select_related("version_added")
+            .select_related("version_removed")
+            .filter(content__ansible_collectionversion__collection_id=OuterRef("pk"))
+            .annotate(
+                last_updated=Greatest(
+                    "version_added__pulp_created", "version_removed__pulp_created"
+                )
+            )
+            .order_by("-last_updated")
+            .only("last_updated")
```

Comment:
This seems overly complicated. RepositoryContent should only get updated when the content is removed (version_removed gets set) so we probably don't need to perform this `select_related`, just check the `pulp_last_updated` on the RepositoryContent.
```suggestion
            RepositoryContent.objects.filter(
                repository_id=repo_version.repository_id,
                version_added__number__lte=repo_version.number,
            )
            .select_related("content__ansible_collectionversion")
            .filter(content__ansible_collectionversion__collection_id=OuterRef("pk"))
            .order_by("-pulp_last_updated")
            .only("pulp_last_updated")
```

Reply:
I tried using `pulp_last_updated` and it doesn't work. That field doesn't seem to get updated when version removed is added to the RepositoryContent entry